src/main/java/io/jenkins/plugins/checks/github/GitSCMChecksContext.java

```diff
@@ -78,12 +87,21 @@ private String getLastBuiltRevisionFromBuildData() {
 
     @Override
     public String getRepository() {
-        String repositoryURL = getUserRemoteConfig().getUrl();
-        if (repositoryURL == null) {
-            return StringUtils.EMPTY;
+        try {
+            String repositoryURL;
+            if (!this.config.getRepository().equals("")) {
+                repositoryURL = getGitCommitEnvironment(this.config.getRepository());
```

Comment:
Using `getGitCommitEnvironment` here is a bit confusing.

I'd be fine with you generifying `getGitCommitEnvironment` to something like `getEnvironmentVariable(...)` or making a `getRepositoryEnvironment()` for it (which contains the `config` logic).